### PR TITLE
MISUV-994 Update the Income Source page

### PR DESCRIPTION
**Pull request by JSimmons072019**

### CODE REVIEW

File: conf/messages [hunk 1]
```diff
@@ -544,7 +544,7 @@ agent.income_source.heading                                     = How does your
 agent.income_source.checkbox.hint                               = Select all that apply
 agent.income_source.selfEmployed                                = Self-employed
 agent.income_source.rentUkProperty                              = Rent out a property in the UK
-agent.income_source.foreignProperty                             = Rent out a foreign property outside the UK
+agent.income_source.foreignProperty                             = Rent out a foreign properties
```
Comment (AlexRimmerHMRC) [inline conf/messages:547]: Is this the correct text? sounds weird being "a" foreign properties

File: test/forms/agent/IncomeSourceFormSpec.scala [hunk 2]
```diff
@@ -21,10 +21,11 @@ import forms.validation.testutils.DataMap.DataMap
 import forms.validation.testutils._
 import models.common.IncomeSourceModel
 import org.scalatest.Matchers._
-import org.scalatestplus.play.{OneAppPerTest, PlaySpec}
+import org.scalatestplus.play.guice.GuiceOneAppPerTest
+import org.scalatestplus.play.PlaySpec
 
 
-class IncomeSourceFormSpec extends PlaySpec with OneAppPerTest {
+class IncomeSourceFormSpec extends PlaySpec with GuiceOneAppPerTest{
```
Comment (AlexRimmerHMRC) [inline test/forms/agent/IncomeSourceFormSpec.scala:28]: I don't think you need the GuiceOneAppPerTest

File: conf/messages (hunk 1 above)
Comment (scullythornton1) [reply conf/messages:547]: that is what we get from https://docs.google.com/spreadsheets/d/14q96g7pIf5LBckAIqIcWCclF4XQefgOyFc5Y6GsEwnk/edit#gid=842503899

File: test/forms/agent/IncomeSourceFormSpec.scala (hunk 2 above)
Comment (scullythornton1) [reply test/forms/agent/IncomeSourceFormSpec.scala:28]: removed.

File: app/connectors/httpparser/GetSubscriptionDetailsHttpParser.scala [hunk 3]
```diff
@@ -29,7 +29,9 @@ object GetSubscriptionDetailsHttpParser {
         response.status match {
           case OK => response.json.validate[T] match {
             case JsSuccess(value, _) => Some(value)
-            case _ => throw new Exception ("Invalid Json for getSubscriptionDetailsHttpReads")
+            case _ => {
+              throw new Exception ("Invalid Json for getSubscriptionDetailsHttpReads")
+            }
```
Comment (AlexRimmerHMRC) [inline app/connectors/httpparser/GetSubscriptionDetailsHttpParser.scala:34]: Can you undo this?

File: it/connectors/stubs/IncomeTaxSubscriptionConnectorStub.scala [hunk 4]
```diff
@@ -30,7 +30,7 @@ object IncomeTaxSubscriptionConnectorStub extends WireMockMethods {
   }
 
   def stubSaveSubscriptionDetails[T](id: String, body: T)(implicit writer: Writes[T]): Unit = {
-    when(method = POST, uri = postUri(id))
+    when(POST, "/income-tax-subscription/self-employments/id/subscriptionId")
```
Comment (AlexRimmerHMRC) [inline it/connectors/stubs/IncomeTaxSubscriptionConnectorStub.scala:33]: Why has this changed?

File: test/views/agent/IncomeSourceViewSpec.scala [hunk 5]
```diff
@@ -78,11 +78,10 @@ class IncomeSourceViewSpec extends ViewSpecTrait {
       page = page(isEditMode = false, addFormErrors = true)
     )
 
-    val testPage = documentCore()
   }
 
   "Display error summary and error notification" when {
-    "foreign property FS is disabled" in {
```
Comment (AlexRimmerHMRC) [inline test/views/agent/IncomeSourceViewSpec.scala:85]: I think these changes are incorrect, from what I see, this part is disabled and the other is enabled

File: test/views/agent/IncomeSourceViewSpec.scala [hunk 6]
```diff
@@ -78,7 +78,6 @@ class IncomeSourceViewSpec extends ViewSpecTrait {
       page = page(isEditMode = false, addFormErrors = true)
     )
 
-    val testPage = documentCore()
```
Comment (AlexRimmerHMRC) [inline test/views/agent/IncomeSourceViewSpec.scala:81]: why did this get removed?

**Pull request merged**